### PR TITLE
[pulsar-client] update to reflect base lib 1.2

### PR DESCRIPTION
**Pull request by felipeguilhermefs**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [subscription type](https://github.com/apache/pulsar-client-node/blob/master/src/ConsumerConfig.cc) and [message properties](https://github.com/apache/pulsar-client-node/blob/master/src/Message.cc)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## pulsar-client ([<kbd>unpkg</kbd>](https://unpkg.com/browse/pulsar-client@latest/))
was missing the following properties:
1. AuthenticationAthenz
1. LogLevel
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f7a2168c9ce512f0ab0667395a5489cbcf727d74
</p>

**Review by bwalendz — CHANGES_REQUESTED** [review]
I think we just get rid of unknown[] here because I think it was done in error. The docs state it's an Object.

File: types/pulsar-client/index.d.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+// Type definitions for pulsar-client 1.2
 // Project: https://github.com/apache/pulsar-client-node
 // Definitions by: Brian Walendzinski <https://github.com/bwalendz>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 3.0
 
 /// <reference types="node" />
 
-export type MessageProperties = unknown[];
+export type MessageProperties = unknown[] | { [key: string ]: string };
```
Comment (bwalendz) [inline types/pulsar-client/index.d.ts:9]: Just use { [key: string]: string }

Comment (felipeguilhermefs) [reply types/pulsar-client/index.d.ts]: Fixed! I just had to change to **interface** instead of **type**, to please the linter

**Review by bwalendz — APPROVED** [review]

**Comment by felipeguilhermefs** [comment]
> Ready to merge

Ready to merge

**Comment by felipeguilhermefs** [comment]
Ready to merge

**Pull request merged**